File: simulator/src/main/java/it/unibo/scafi/simulation/gui/model/SimulationManager.java
package it.unibo.scafi.simulation.gui.model;

import it.unibo.scafi.simulation.gui.Simulation;

/**
 * Created by chiara on 14/11/16.
 */
public interface SimulationManager {

    Simulation getSimulation();
    void setSimulation(Simulation simulation);
    void setPauseFire(Double pauseFire);
    void start();
    void resume();
    void stop();
    void pause();
    void step(int num_step);
}

File: simulator-gui/src/main/scala/it/unibo/scafi/simulation/gui/model/SimulationManager.scala
package it.unibo.scafi.simulation.gui.model

import it.unibo.scafi.simulation.gui.Simulation

trait SimulationManager {
  var simulation: Simulation

  def setPauseFire(pauseFire: Double)

  def start()

  def resume()

  def stop()

  def pause()

  def step(num_step: Int)
}
